service/trino-verifier/src/main/java/io/trino/verifier/Validator.java

```diff
@@ -539,13 +545,13 @@ private QueryResult executeQuery(String url, String username, String password, Q
                 exception = (Exception) e.getCause();
             }
             State state = isPrestoQueryInvalid(e) ? State.INVALID : State.FAILED;
-            return new QueryResult(state, exception, null, null, queryId, ImmutableList.of());
+            return new QueryResult(state, exception, nanosSince(start), queryCpuTime, queryId, ImmutableList.of());
         }
         catch (VerifierException e) {
-            return new QueryResult(State.TOO_MANY_ROWS, e, null, null, queryId, ImmutableList.of());
+            return new QueryResult(State.TOO_MANY_ROWS, e, nanosSince(start), queryCpuTime, queryId, ImmutableList.of());
         }
         catch (UncheckedTimeoutException e) {
-            return new QueryResult(State.TIMEOUT, e, null, null, queryId, ImmutableList.of());
+            return new QueryResult(State.TIMEOUT, e, nanosSince(start), queryCpuTime, queryId, ImmutableList.of());
```

Comment:
those look unrelated - could be separate commit.

Reply:
This is a part of the `Record query id for failed queries in verifier`. Maybe i should change the commit message to something like `Record query id and runtime information for failed queries in verifier`?